### PR TITLE
Add support for LoongArch64

### PR DESCRIPTION
**Pull request by donmor**
Yet another architecture :-)
I don' know much about rust, but at least it built on qemu <:-/

### CODE REVIEW

**Comment by xiangzhai** [comment]
Hi @donmor 

Thanks for your contribution for LoongArch!

```
WARNING: The requested image's platform (linux/loong64) does not match the detected host platform (linux/amd64) and no specific platform was requested
exec /usr/sbin/sh: exec format error
```

Does it need to cross compile for loongarch64 target on x86_64 host?

Thanks,
Leslie Zhai

**Comment by donmor** [comment]
> Hi @donmor
> 
> Thanks for your contribution for LoongArch!
> 
> ```
> WARNING: The requested image's platform (linux/loong64) does not match the detected host platform (linux/amd64) and no specific platform was requested
> exec /usr/sbin/sh: exec format error
> ```
> 
> Does it need to cross compile for loongarch64 target on x86_64 host?
> 
> Thanks, Leslie Zhai

I think so but I failed to make it pass CI. I can build this locally on qemu vm, though <:-/

**Comment by donmor** [comment]
And I'm not sure if it is finely optimized. It's just a simpliest patch to build this package on loongarch.

**Comment by heiher** [comment]
Could we skip CI and just add basic support like `ioctl` for now? We can add CI at a later time, such as when promoting to Tier 2.

**Comment by chenx97** [comment]
loongarch64 is already in tier2. This PR should get more attention.

**Comment by MingcongBai** [comment]
> loongarch64 is already in tier2. This PR should get more attention. @asomers would you like to take a look?

Seconded. The user base of this architecture is definitely picking up and `nix` is an essential dependency for most Rust applications.

File: .cirrus.yml [hunk 1]
```diff
@@ -105,6 +105,9 @@ task:
     - name: Linux i686 musl
       env:
         TARGET: i686-unknown-linux-musl
+    - name: Linux LoongArch64
```
Comment (SteveLauC) [inline .cirrus.yml:108]: Please add this here, Nix has switched to GitHub action for most CI checks:

 https://github.com/nix-rust/nix/blob/50e4283b35f3f34e138d138fd889f7e3c424a5c2/.github/workflows/ci.yml#L94

File: Cross.toml [hunk 2]
```diff
@@ -3,3 +3,5 @@ passthrough = [
 	"RUSTFLAGS",
 	"RUST_TEST_THREADS"
 ]
+[target.loongarch64-unknown-linux-gnu]
+image = "ghcr.io/loongarchlinux/archlinux:latest"
```
Comment (SteveLauC) [inline Cross.toml:7]: Looks like Loongarch has been [supported by cross](https://github.com/cross-rs/cross?tab=readme-ov-file#supported-targets), is this custom image still needed?

File: src/sys/ioctl/linux.rs [hunk 3]
```diff
@@ -41,6 +41,7 @@ mod consts {
     target_arch = "s390x",
     target_arch = "x86_64",
     target_arch = "aarch64",
+    target_arch = "loongarch64",
```
Comment (SteveLauC) [inline src/sys/ioctl/linux.rs:44]: Could you guys please check what are currently supported and what are not? And, if there is still a lot of work to do, and you guys don't mind, a tracking issue would be appreciated:)

**Comment by SteveLauC** [comment]
The author of this PR hasn't been active on GitHub for a while, so feel free to pick it up.

**Comment by xen0n** [comment]
Triage: https://github.com/nix-rust/nix/pull/2045 (since `nix 0.27.0`) already contained the necessary ioctl change, so if this PR is to be rebased, only the CI changes will remain. `nix` is already working on LoongArch from `0.27.0` on, if one doesn't intend to hack on `nix` itself.

**Comment by SteveLauC** [comment]
> I'm not sure if cirrus.yml is still necessary, so I'll leave this open but remove unnecessary changes.

This is no longer needed, close it as superseded by #2461.

**Pull request closed without merging**